core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -1004,7 +1004,9 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
           recordsWritten += 1
         }
       } finally {
-        writer.close(hadoopContext)
+        if (writer != null) {
```

Comment:
Why check here? if it's `null` then this fails already in the loop. Just assert it's not `null` with `require` after it's obtained, if this is supposed to die in this case anyway.